### PR TITLE
refactor(cz): better typing and refactor message method

### PR DESCRIPTION
**Pull request by bearomorphism**
## Checklist

- [ ] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly
  - [ ] Check if there are any broken links in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: commitizen/cz/conventional_commits/conventional_commits.py [hunk 1]
```diff
@@ -147,24 +157,22 @@ def questions(self) -> Questions:
...
 
-    def message(self, answers: dict) -> str:
-        prefix = answers["prefix"]
-        scope = answers["scope"]
-        subject = answers["subject"]
-        body = answers["body"]
-        footer = answers["footer"]
-        is_breaking_change = answers["is_breaking_change"]
+    def message(self, answers: ConventionalCommitsAnswers) -> str:  # type: ignore[override]
+        parts: list[str] = [answers["prefix"]]
```
Comment (Lee-W) [inline commitizen/cz/conventional_commits/conventional_commits.py:161]: I believe the new logic makes it harder for humans to read. 🤔

**Review by Lee-W — COMMENTED** [review]
mostly good, one nitpick

File: commitizen/cz/conventional_commits/conventional_commits.py (hunk 1 above)
Comment (bearomorphism) [reply commitizen/cz/conventional_commits/conventional_commits.py]: reverted. agree

**Review by Lee-W — APPROVED** [review]

**Comment by Lee-W** [comment]
opps, merge conflict

**Comment by bearomorphism** [comment]
@Lee-W could you help to rerun the check?
